### PR TITLE
[MM-29295] Fix data race in app.CreatePost()

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR fixes a data race caused by sharing the same `*model.Post` between goroutines. We now make a copy
This case is covered by `TestHookMessageHasBeenPosted` which would fail when run with `-race`.

#### Ticket

https://mattermost.atlassian.net/browse/MM-29295

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: app/post.go [hunk 1]
```diff
@@ -308,11 +308,13 @@ func (a *App) CreatePost(post *model.Post, channel *model.Channel, triggerWebhoo
 	// might be duplicating requests.
 	a.Srv().seenPendingPostIdsCache.SetWithExpiry(post.PendingPostId, rpost.Id, PENDING_POST_IDS_CACHE_TTL)
 
+	// We make a copy of the post for the plugin hook to avoid a race condition.
+	postCopy := rpost.Clone()
```
Comment (agnivade) [inline app/post.go:312]: 0/5, perhaps `rPostCopy` makes it clearer.

**Review by agnivade — APPROVED** [review]

**Comment by streamer45** [comment]
@lindalumitchell 

This PR fixes an existing race condition triggered by the `MessageHasBeenPosted` plugin hook.
It does not alter functionality and the case is covered by unit tests. I've also left QA steps to verify this.

**Review by prapti — APPROVED** [review]
Tested on local server using demo plugin that registers the `MessageHasBeenPosted` hook. There is no race related error noted in the Jira ticket.
Approving.
P.s.: Thank you for the patience through all my questions @streamer45!

**Pull request merged**